manifests/init.pp

```diff
@@ -55,18 +52,32 @@
     }
 
     if $::operatingsystem == 'SLES' and $::operatingsystemmajrelease == '10' {
-      $_package_file_name = "${puppet_agent::package_name}-${_package_version}-1.sles10.${::architecture}.rpm"
-    } elsif $::operatingsystem == 'Solaris' and $::operatingsystemmajrelease == '10' {
+      $_package_file_name = "${puppet_agent::package_name}-${package_version}-1.sles10.${::architecture}.rpm"
+    } elsif $::operatingsystem == 'Solaris' {
       if $arch =~ /^sun4[uv]$/ {
-        $_package_file_name = "${puppet_agent::package_name}-${_package_version}-1.sparc.pkg.gz"
+        if $::operatingsystemmajrelease == '10' {
+          $_package_file_name = "${puppet_agent::package_name}-${package_version}-1.sparc.pkg.gz"
+        } elsif $::operatingsystemmajrelease == '11' {
+          # Strip letters from development builds.
+          $_version_without_letters = regsubst($package_version, /[a-zA-Z]/, '', 'G')
+          $_solaris_version = regsubst($_version_without_letters, /(^-|-$)/, '', 'G')
+          $_package_file_name = "${puppet_agent::package_name}@${_solaris_version},5.11-1.sparc.p5p"
+        }
       } else {
-        $_package_file_name = "${puppet_agent::package_name}-${_package_version}-1.i386.pkg.gz"
+        if $::operatingsystemmajrelease == '10' {
+          $_package_file_name = "${puppet_agent::package_name}-${package_version}-1.i386.pkg.gz"
+        } elsif $::operatingsystemmajrelease == '11' {
+          # Strip letters from development builds.
+          $_version_without_letters = regsubst($package_version, /[a-zA-Z]/, '', 'G')
+          $_solaris_version = regsubst($_version_without_letters, /(^-|-$)/, '', 'G')
+          $_package_file_name = "${puppet_agent::package_name}@${_solaris_version},5.11-1.i386.p5p"
```

Comment:
Couldn't the i386 vs sparc part be set in a variable? Then we wouldn't need to duplicate all the regex for both platforms.